### PR TITLE
bin: use SIGUSR1 to dump metrics.

### PR DESCRIPTION
**Pull request by siddontang**
Use SIGUSR1 to dump the metrics.

@ngaut @BusyJay @hhkbp2 @overvenus

### CODE REVIEW

File: src/bin/tikv-server.rs [hunk 1]
```diff
@@ -613,15 +615,22 @@ fn start_server<T, S>(mut server: Server<T, S>, mut el: EventLoop<Server<T, S>>)
...
                 info!("receive signal {}, stopping server...", sig);
                 ch.send(Msg::Quit).unwrap();
                 break;
             }
+            SIGUSR1 => {
+                let mut buffer = vec![];
+                let metric_familys = prometheus::gather();
+                let encoder = TextEncoder::new();
+                encoder.encode(&metric_familys, &mut buffer).unwrap();
+                info!("{}", String::from_utf8(buffer.clone()).unwrap());
```
Comment (overvenus) [inline src/bin/tikv-server.rs]: Why `clone()` ? How about `std::str::from_utf8`

File: src/bin/tikv-server.rs [hunk 2]
```diff
@@ -613,15 +615,22 @@ fn start_server<T, S>(mut server: Server<T, S>, mut el: EventLoop<Server<T, S>>)
...
+    use nix::sys::signal::{SIGTERM, SIGINT, SIGUSR1};
+    let trap = Trap::trap(&[SIGTERM, SIGINT, SIGUSR1]);
     for sig in trap {
         match sig {
             SIGTERM | SIGINT => {
                 info!("receive signal {}, stopping server...", sig);
                 ch.send(Msg::Quit).unwrap();
                 break;
             }
+            SIGUSR1 => {
```
Comment (overvenus) [inline src/bin/tikv-server.rs]: Later, we should add some docs for triggering metrics dump.

File: src/bin/tikv-server.rs (hunk 1 above)
Comment (siddontang) [inline src/bin/tikv-server.rs]: Em, I just copy the previous code, will fix it.

**Comment by ngaut** [comment]
LGTM

**Pull request merged**